src/modules/Dashboard/features/RespondentDetails/RespondentDetails.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { StyledBody } from 'shared/styles';
+import { workspaces } from 'shared/state';
+import { applets, users } from 'modules/Dashboard/state';
+import { useAppDispatch } from 'redux/store';
+import { getRespondentDetails } from 'modules/Dashboard/state/Users/Users.thunk';
+import { useRespondentDetailsTabs } from 'modules/Dashboard/pages/RespondentDetails/RespondentDetails.hooks';
+import { LinkedTabs, Spinner } from 'shared/components';
+import { palette } from 'shared/styles/variables/palette';
+
+import { StyledTitleContainer } from './RespondentDetails.styles';
+
+export const RespondetDetails = () => {
+  const dispatch = useAppDispatch();
+  const respondentTabs = useRespondentDetailsTabs();
+  const { appletId, respondentId } = useParams();
+  const { ownerId } = workspaces.useData() || {};
+  const { useRespondent } = users;
+  const respondent = useRespondent();
+
+  useEffect(() => {
+    if (!appletId) return;
+    const { getEvents } = applets.thunk;
+    dispatch(getEvents({ appletId, respondentId }));
+
+    if (!respondentId || !ownerId) return;
```

Comment:
In my opinion, it would be better to transfer ``!respondentId`` to the first early return.